### PR TITLE
Feature/improve documentation

### PR DESCRIPTION
**Pull request by arbulu89**
I have reformatted most of the documentation as during the evolution of the project, it was getting obsolete and too confusing. 
I have tried to simplify the documentation, removing technical aspects of the code. I think that most of that texts were totally redundant and were not providing any good information.

I have just pointed the documentation to the relevant `variables` files, where everything should be documented and reachable/understandable by everyone. I don't think duplicating the descriptions in a readme really helps.

https://github.com/SUSE/ha-sap-terraform-deployments/issues/415

### CODE REVIEW

File: aws/README.md [hunk 1]
```diff
@@ -134,88 +134,17 @@ The infrastructure deployed includes:
 
 By default it creates 3 instances in AWS: one for support services (mainly iSCSI as most other services - DHCP, NTP, etc - are provided by Amazon) and 2 cluster nodes, but this can be changed to deploy more cluster nodes as needed.
 
-## Provisioning by Salt
-By default, the cluster and HANA installation is done using Salt Formulas in foreground.
-To customize this provisioning, you have to create the pillar files (cluster.sls and hana.sls) according to the examples in the [pillar_examples](../pillar_examples) folder (more information in the dedicated [README](../pillar_examples/README.md))
+# Specifications
 
-# Specification:
+In order to deploy the environment, different configurations are available through the terraform variables. This variables can be configured using a `terraform.tfvars` file. An example is available in [terraform.tfvars.example](./terraform.tvars.example). To find all the available variables check the [variables.tf](./variables.tf) file.
```
Comment (juadk) [inline aws/README.md:139]:
```suggestion
In order to deploy the environment, different configurations are available through the terraform variables. These variables can be configured using a `terraform.tfvars` file. An example is available in [terraform.tfvars.example](./terraform.tvars.example). To find all the available variables check the [variables.tf](./variables.tf) file.
```

File: azure/README.md [hunk 2]
```diff
@@ -85,92 +85,17 @@ By default, this configuration will create 3 virtual machines in Azure: one for
...
-Find more information about the hana and cluster formulas in (check the pillar.example files):
-
--   [https://github.com/SUSE/saphanabootstrap-formula](https://github.com/SUSE/saphanabootstrap-formula)
--   [https://github.com/SUSE/habootstrap-formula](https://github.com/SUSE/habootstrap-formula)
-
-As a good example, you could find some pillar examples into the folder [pillar_examples](../pillar_examples)
-These files **aren't ready for deployment**, be careful to customize them or create your own files.
+# Specifications
+
+In order to deploy the environment, different configurations are available through the terraform variables. This variables can be configured using a `terraform.tfvars` file. An example is available in [terraform.tfvars.example](./terraform.tvars.example). To find all the available variables check the [variables.tf](./variables.tf) file.
```
Comment (juadk) [inline azure/README.md:90]:
```suggestion
In order to deploy the environment, different configurations are available through the terraform variables. These variables can be configured using a `terraform.tfvars` file. An example is available in [terraform.tfvars.example](./terraform.tvars.example). To find all the available variables check the [variables.tf](./variables.tf) file.
```

File: gcp/README.md [hunk 3]
```diff
@@ -67,92 +67,17 @@ The infrastructure deployed includes:
 
 By default, this configuration will create 3 instances in GCP: one for support services (mainly iSCSI as most other services - DHCP, NTP, etc - are provided by Google) and 2 cluster nodes, but this can be changed to deploy more cluster nodes as needed.
 
-## Provisioning by Salt
-By default, the cluster and HANA installation is done using Salt Formulas in foreground.
-To customize this provisioning, you have to create the pillar files (cluster.sls and hana.sls) according to the examples in the [pillar_examples](https://github.com/SUSE/ha-sap-terraform-deployments/blob/master/pillar_examples) folder (more information in the dedicated [README](https://github.com/SUSE/ha-sap-terraform-deployments/blob/master/pillar_examples/README.md))
+# Specifications
 
-## Specification
+In order to deploy the environment, different configurations are available through the terraform variables. This variables can be configured using a `terraform.tfvars` file. An example is available in [terraform.tfvars.example](./terraform.tvars.example). To find all the available variables check the [variables.tf](./variables.tf) file.
```
Comment (juadk) [inline gcp/README.md:72]:
```suggestion
In order to deploy the environment, different configurations are available through the terraform variables. These variables can be configured using a `terraform.tfvars` file. An example is available in [terraform.tfvars.example](./terraform.tvars.example). To find all the available variables check the [variables.tf](./variables.tf) file.
```

File: libvirt/README.md [hunk 4]
```diff
@@ -42,101 +42,17 @@ terraform apply
...
-If the current *main.tf* is used, only *uri* (usually SAP HANA cluster deployment needs a powerful machine, not recommended to deploy locally) and *hana_inst_media* parameters must be updated.
-
-* hana.sls
-
-**hana.sls** is used to configure the SAP HANA cluster. Check the options in: [saphanabootstrap-formula](https://github.com/SUSE/saphanabootstrap-formula)
-
-* cluster.sls
-
-**cluster.sls** is used to configure the HA cluster. Check the options in: [habootstrap-formula](https://github.com/SUSE/habootstrap-formula)
+In order to deploy the environment, different configurations are available through the terraform variables. This variables can be configured using a `terraform.tfvars` file. An example is available in [terraform.tfvars.example](./terraform.tvars.example). To find all the available variables check the [variables.tf](./variables.tf) file.
```
Comment (juadk) [inline libvirt/README.md:47]:
```suggestion
In order to deploy the environment, different configurations are available through the terraform variables. These variables can be configured using a `terraform.tfvars` file. An example is available in [terraform.tfvars.example](./terraform.tvars.example). To find all the available variables check the [variables.tf](./variables.tf) file.
```

File: pillar_examples/README.md [hunk 5]
```diff
@@ -1,49 +1,42 @@
...
 
-      `cp pillar_examples/automatic/hana/*.sls salt/hana_node/files/pillar/`
+```
+cp pillar_examples/automatic/hana/*.sls salt/hana_node/files/pillar/
+cp pillar_examples/automatic/drbd/*.sls salt/drbd_node/files/pillar/
+cp pillar_examples/automatic/netweaver/*.sls salt/netweaver_node/files/pillar/
+```
 
-  - For a customize and production environment, you must use pillar files which are in your choosen [provider directory](../pillar_examples) (AWS, Azure, GCP, Libvirt).
+Besides this option, the `terraform.tfvars` `pre_deployment` variable will execute this operations if it's enabled before the deployment.
```
Comment (juadk) [inline pillar_examples/README.md:19]:
```suggestion
Besides this option, the `terraform.tfvars` `pre_deployment` variable will execute these operations if it's enabled before the deployment.
```

**Comment by arbulu89** [comment]
@juadk Yes, having a description should be mandatory. I will check the missing entries, and improve others if needed

**Comment by arbulu89** [comment]
@juadk I will merge this PR, and I'm working in other with all of the descriptions included. This PR is already big enough that I prefer to have a new one.

**Pull request merged**